source/core.py

```diff
@@ -79,8 +79,21 @@ def doStartupDialogs():
 		gui.messageBox(_("Your gesture map file contains errors.\n"
 				"More details about the errors can be found in the log file."),
 			_("gesture map File Error"), wx.OK|wx.ICON_EXCLAMATION)
-	if not globalVars.appArgs.secure and not config.isAppX and not config.conf['update']['askedAllowUsageStats']:
-		gui.runScriptModalDialog(gui.AskAllowUsageStatsDialog(None))
+	try:
+		import updateCheck
+	except RuntimeError:
+		updateCheck=None
+	if updateCheck and not config.conf['update']['askedAllowUsageStats']:
+		# a callback to save config after the usage stats question dialog has been answered.
+		def onResult(ID):
+			import wx
+			if ID in (wx.ID_YES,wx.ID_NO):
+				try:
+					config.conf.save()
+				except:
+					pass

```

Comment:
What exception are we likely to get here? Can we at least log it? If there is an expected exception, then we should be able to specify what it is, and probably also add a log message.

Reply:
The exception will be if we're on a read-only file system. To be honest I'm not sure at all what type it is as  the save method itself catches a generic Exception itself.  It also logs a warning, so there is probably no need for us to log again.

Reply:
At the very least we know that it is `Exception`. Can you catch that and pass. Generally, it would be nice to tighten this up so that we are not inadvertently burying other errors. 

It looks like there could be:
- `raise ctypes.WinError()` from `def FaultTolerantFile(name)`
- `BaseException` from `def NamedTemporaryFile`

Other than that there is whatever could be raised by the contents of `configObj.write`

This is quite hard to be sure we have the full list, so I'll leave it up to you to decide what to do.